sdk/communication/azure-communication-identity/tests/test_communication_identity_client.py

```diff
@@ -8,42 +8,35 @@
 from datetime import timedelta
 from azure.communication.identity import CommunicationIdentityClient
 from azure.communication.identity import CommunicationTokenScope
-from azure.core.credentials import AccessToken
-from devtools_testutils import is_live
-from _shared.helper import URIIdentityReplacer, URIMsalUsernameReplacer
-from _shared.testcase import BodyReplacerProcessor
-from testcase import CommunicationIdentityTestCase
+from devtools_testutils import AzureRecordedTestCase, is_live, recorded_by_proxy
 from _shared.communication_service_preparer import CommunicationPreparer
 from _shared.utils import get_http_logging_policy
-from utils import is_token_expiration_within_allowed_deviation
+from _shared.fake_token_credential import FakeTokenCredential
+from utils import is_token_expiration_within_allowed_deviation, generate_teams_user_aad_token, \
+    skip_get_token_for_teams_user_test
 from azure.identity import DefaultAzureCredential
 from azure.communication.identity._shared.utils import parse_connection_str
-from parameterized import parameterized
-class FakeTokenCredential(object):
-    def __init__(self):
-        self.token = AccessToken("Fake Token", 0)
-
-    def get_token(self, *args):
-        return self.token
-
-class CommunicationIdentityClientTest(CommunicationIdentityTestCase):
-    def setUp(self):
-        super(CommunicationIdentityClientTest, self).setUp()
-        self.recording_processors.extend([
-            BodyReplacerProcessor(keys=["id", "token", "appId", "userId", "domain_name"]),
-            URIIdentityReplacer(),
-            URIMsalUsernameReplacer()])
-    
+from acs_identity_decorator import acs_identity_decorator
+
+
+class ArgumentPasser:
+    def __call__(self, fn):
+        def _wrapper(test_class, _, value, **kwargs):
+            fn(test_class, _, value, **kwargs)
+        return _wrapper
+
+class TestClient(AzureRecordedTestCase):
     @CommunicationPreparer()
+    @recorded_by_proxy
     def test_create_user_from_managed_identity(self, communication_livetest_dynamic_connection_string):
```

Comment:
I was wondering whether any test parameters(here I mean `communication_livetest_dynamic_connection_string`) it should be in the kwargs as explained in here [new test structure](https://github.com/Azure/azure-sdk-for-python/blob/main/doc/dev/test_proxy_migration_guide.md#new-test-structure) :
`So, arguments that are passed to a test by a preparer -- for example, example_variable in test_example_with_preparer above -- should be accepted via **kwargs and popped off at the start of a test.`

Reply:
That is a good point -- using the `**kwargs` structure is more future-proof. Using positional parameters for test parameters isn't compatible with pytest fixtures, which are ultimately what we'd like to use in tests in place of decorators like `@recorded_by_proxy`.

Reply:
Thank you for pointing this out. I refactored not to use test parameters like this.